xarray/core/rolling.py

```diff
@@ -205,7 +205,10 @@ def reduce(self, func, **kwargs):
                    for _, window in self]
 
         # Find valid windows based on count
-        concat_dim = self.window_labels if self.dim in self.obj else self.dim
+        if self.dim in self.obj.coords:
+            concat_dim = self.window_labels
+        else:
+            concat_dim = self.dim
```

Comment:
Was this the only place we were looking for coords in this way?

Reply:
There were a couple more. Verified by converting the warning into an error and running the test suite.